tibiawikisql/models/item.py

```diff
@@ -282,15 +282,18 @@ def from_article(cls, article):
             if attribute in item._raw_attributes and item._raw_attributes[attribute]:
                 item.attributes.append(ItemAttribute(item_id=item.article_id, name=name,
                                                      value=item._raw_attributes[attribute]))
-        item._parse_attributes()
-        item._parse_resists()
-        vocations = item._raw_attributes.get('vocrequired')
-        if vocations and "none" not in vocations.lower():
-            vocation = vocations.replace('and', '+').replace(',', '+').replace(' ', '')
-            item.attributes.append(ItemAttribute(item_id=item.article_id, name="vocation", value=vocation))
-        item._parse_sounds()
-        item._parse_store_value()
-        return item
+        try:
+            item._parse_attributes()
+            item._parse_resists()
+            vocations = item._raw_attributes.get('vocrequired')
+            if vocations and "none" not in vocations.lower():
+                vocation = vocations.replace('and', '+').replace(',', '+').replace(' ', '')
+                item.attributes.append(ItemAttribute(item_id=item.article_id, name="vocation", value=vocation))
+            item._parse_sounds()
+            item._parse_store_value()
+            return item
+        except Exception:
+            return None
```

Comment:
Wouldn't this make the entire item not be parsed if something failed when parsing attributes / resists/ sounds / store value?

Probably need to add some validations inside _parse_store_value to handle the case that was raising the exception.

Reply:
That's what I had done at first, but added this try/catch block here instead to also catch exceptions coming from attributes/resists/sounds/store value all at once.

Returning None here will include the failed items in the "unparsed" array from tibiawikisql/__main__.py:generate, so they are printed in stdout along with the other items that failed for whatever reason.

Reply:
But for example ,if an item is all fine, but it has an incorrect value in the store section, it would make all of the other information be dropped, wouldn't it?

Reply:
Yes, the reasoning behind it is that any untreated exception raised by any of those functions, like the one I found on _parse_store_value, could cause the script to abort.

I guessed that the behavior of throwing away individual items' data and logging that on stdout would be preferable over violently halting the entire script, but we could take the other path and insert checks on each individual function instead of just enclosing all of them in this one.

In the latter case, I will remove this try/catch block from here and put it in the _parse_store_value function, which is the one that was causing the script to fail this time.

Reply:
Hmmm you're right about wanting to avoid halting the entire script. Maybe the try catch should be on the script side?

Because this is also a library that can be imported and used to fetch live data from TibiaWiki, for example, you can get the content of an item's article and parse it as an Item, in this case, the `from_article` function itself, should it just return `None`, or should it throw the exception if something went wrong? So you can see that the article itself is wrong?

Reply:
The point of failure is inside `Item._parse_store_value`, which is called by `Item.from_article`, so we must throw the exception in this class. The exception is, in this case, both being thrown and returning `None`, because the caller (`__main.py__.generate`) knows what to do with `None`.

The way it is right now (including my changes) produces this output:
```
Parsing items ██████████ 7385/7385
Could not parse 9 articles.
        -> Knightly Bed,Knightly Bench,Ladder,Natural Products,Objects (Broken Dream),Quest Objects,Refuse,Trophies (Objects),Underground Plants
        Parsed articles in 57.35 seconds.
```

It tells you which items returned `None`, but doesn't tell you why. Knightly Bed and Bench weren't parsed in this case because the article's infobox has an empty store price, so my change caught that exception and returned `None`. For the other items' pages, they simply don't have an infobox, so there is nothing `from_article` can parse, also returning `None`. Maybe those articles shouldn't be included in the parsing list?

I think logging the reason why each item failed to parse would be a good idea (maybe along with a `-v, --verbose` arg to avoid flooding the logs). In this case, I'd say `Item.from_article` should only raise an exception for `__main.py__.generate` to handle.

Reply:
Yeah, I think a -v option would be good. It is tricky, because the reasons would have to be stored and then printed at the end, otherwise, it breaks the progress bar.